cmds/core/netcat/connect_test.go

```diff
@@ -20,6 +22,110 @@ import (
 	"github.com/u-root/u-root/pkg/netcat"
 )
 
+func TestConnectMode(t *testing.T) {
+	connectErr := errors.New("failed to connect")
+
+	tests := []struct {
+		name          string
+		network       string
+		address       string
+		connectFunc   func(output io.Writer, network, address string) error
+		expectedError error
+	}{
+		{
+			name:    "TCPv4 success",
+			network: "tcp",
+			address: "localhost:8080",
+			connectFunc: func(output io.Writer, network, address string) error {
+				if network == "tcp4" {
+					return nil
+				}
+				return connectErr
+			},
+			expectedError: nil,
+		},
+		{
+			name:    "TCPv6 success",
+			network: "tcp",
+			address: "localhost:8080",
+			connectFunc: func(output io.Writer, network, address string) error {
+				if network == "tcp6" {
+					return nil
+				}
+				return connectErr
+			},
+			expectedError: nil,
+		},
+		{
+			name:    "UDPv4 success",
+			network: "udp",
+			address: "localhost:8080",
+			connectFunc: func(output io.Writer, network, address string) error {
+				if network == "udp4" {
+					return nil
+				}
+				return connectErr
+			},
+			expectedError: nil,
+		},
+		{
+			name:    "UDPv6 success",
+			network: "udp",
+			address: "localhost:8080",
+			connectFunc: func(output io.Writer, network, address string) error {
+				if network == "udp6" {
+					return nil
+				}
+				return connectErr
+			},
+			expectedError: nil,
+		},
+		{
+			name:    "TCPv4 and TCPv6 failure",
+			network: "tcp",
+			address: "localhost:8080",
+			connectFunc: func(output io.Writer, network, address string) error {
+				return connectErr
+			},
+			expectedError: connectErr,
+		},
+		{
+			name:    "UDPv4 and UDPv6 failure",
+			network: "udp",
+			address: "localhost:8080",
+			connectFunc: func(output io.Writer, network, address string) error {
+				return connectErr
+			},
+			expectedError: connectErr,
+		},
+		{
+			name:    "Other network success",
+			network: "unix",
+			address: "/tmp/socket",
+			connectFunc: func(output io.Writer, network, address string) error {
+				if network == "unix" {
+					return nil
+				}
+				return connectErr
+			},
+			expectedError: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cmd := &cmd{
+				connectStubFn: tt.connectFunc,
+			}
+			err := cmd.connectMode(io.Discard, tt.network, tt.address)
+			if !errors.Is(err, tt.expectedError) {
+				t.Errorf("Expected error %v, got %v", tt.expectedError, err)
```

Comment:
The common pattern is the 'got, want' pattern, I believe, i.e. what you got, and what you expected. 
Certainly in my code reviews at Google that's what they'd tell me. 

It makes it easier to scan the logs in Ci if we keep this consistent. again, I leave it to you.

Reply:
Agree, consistency matters.